src/main/java/org/broadinstitute/hellbender/utils/IntToDoubleFunctionCache.java

```diff
@@ -0,0 +1,70 @@
+package org.broadinstitute.hellbender.utils;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A helper class to maintain a cache of an int to double function
+ * The cache expands when a number is not available.
+ * NOTE: this cache is thread safe and it may be accessed from multiple threads.
+ */
+public abstract class IntToDoubleFunctionCache {
+    private static final Logger logger = LogManager.getLogger(IntToDoubleFunctionCache.class);
+
+    private double[] cache = new double[] { };
+
+    protected abstract int maxSize();
+
+    protected abstract double compute(final int n);
+
+    public IntToDoubleFunctionCache() { }
+
+    /**
+     * Get the value of log10(n), expanding the cache as necessary
+     * @param i operand
+     * @return log10(n)
+     */
+    public double get(final int i) {
+        Utils.validateArg(i >= 0, () -> String.format("Cache doesn't apply to negative number %d", i));
```

Comment:
maybe document this restriction (or even change the class name)

Reply:
done